{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/cli.py

```diff
@@ -6,136 +6,159 @@
 
-def sanitize(loc: str) -> (bool, Location):
-    loc_proper = loc.strip()
-    if loc_proper.startswith('#'):
-        loc_proper = loc_sanitized = ''
-    else:
-        loc_sanitized = loc_proper
-        loc_sanitized = re.sub("[- ,.&']", '_', loc_sanitized).lower()
-    return bool(loc_proper), Location(loc_proper, loc_sanitized)
+    >>> sanitize("Cote d'Ivoire")
+    Location(proper="Cote d'Ivoire", sanitized="cote_d-ivoire")
 
+    """
+    # TODO: Check that they're in a call to get_location_id from
+    #    vivarium_gbd_access.gbd.  Not doing this now because it's not
+    #    specified as a proper dependency in the setup.py and would be a
+    #    bit of a pain to do now.
+    for location in locations:
+        proper = location.strip()
+        sanitized = re.sub("[- ,.&']", '_', proper).lower()
+        yield Location(proper, sanitized)
+
+
+def parse_locations(locations_file: Optional[str], single_location: Optional[str]) -> List[str]:
+    """Parses location inputs into a list of location strings.
+
+    If no arguments are provided, this will default to the ``locations.txt``
+    file located in the repository model_specifications directory.
+
+    Parameters
+    ----------
+    locations_file
+        Path to a file containing a list of locations to generate model
+        specifications for.
+    single_location
+        Optional single location to generate a model specification for.
+
+    Returns
+    -------
+        A list of location strings.
+
+    Raises
+    ------
+    ValueError
+        If both ``locations_file`` and ``single_loc`` are provided or if
+        a ``locations_file`` with no locations is provided.
 
-def get_sanitized_locations(single_loc: str, loc_file: Path) -> List[Location]:
-    """ Naming convention is to lowercase the location string, replace spaces with underscores,
-        and capitalize the first letter. No manipulation happens in the template code.
     """
-    def helper(item):
-        valid, loc = sanitize(item)
-        if valid:
-            locs.append(loc)
-
-    locs = []
-    if single_loc:
-        helper(single_loc)
-    else:
-        with loc_file.open('r') as infile:
-            for line in infile:
-                helper(line)
-    return locs
-
-
-def log_and_exit(f: Callable) -> Callable:
-    @wraps(f)
-    def wrapper(*args, **kwds):
-        f(*args, **kwds)
-        exit(1)
-    return wrapper
-
-    
-def verify_args(template: Path, locations_file: Path, single_location: str, output_dir: Path) -> None:
-    exit_logger = log_and_exit(logger.error)
-    if not template.exists():
-        exit_logger(f'\nError: the template file "{template}" does not exist.')
-    if not locations_file.exists():
-        exit_logger(f'\nError: the locations file "{locations_file}" does not exist. Create locations.txt '
-                    'or use the "-s" switch to specify a single location.')
-    if not single_location and not locations_file.stat().st_size:
```

Comment:
This was the only logic error, I think.  We would have passed if we were provided no location and a location file full of comments but no actual text.